### PR TITLE
Fixing MCS Integration Tests

### PR DESCRIPTION
**Pull request by ThomasSchellenbergNextCentury**
- Fixed the integration tests broken by PR https://github.com/NextCenturyCorporation/MCS/pull/462
- Fixed a bug in which we weren't handling numbers correctly when comparing the user's input parameters to the action restrictions (comparing numeric strings, ints, and floats).

### CODE REVIEW

**Review by jaudick — APPROVED** [review]

File: machine_common_sense/controller.py [hunk 1]
```diff
@@ -283,25 +295,22 @@ def step(self, action: str, **kwargs) -> Optional[StepMetadata]:
...
-            logger.warning(
+            logger.error(
                 f"The given action '{action}' with parameters "
                 f"'{kwargs}' isn't in the action_list. Ignoring your action. "
                 f"Please call controller.step() with an action in the "
                 f"action_list. Possible actions at step {self.__step_number}:"
             )
             for action_data in action_list:
-                logger.warning(f'    {action_data}')
+                logger.error(f'    {action_data}')
```
Comment (kdrumm) [inline machine_common_sense/controller.py:313]: Note to others, this change will cause an additional log when running the unit tests, specifically 'run_restricted_action_list_test'

**Review by kdrumm — APPROVED** [review]
I got the following log
``

File: machine_common_sense/controller.py (hunk 1 above)
Comment (ThomasSchellenbergNextCentury) [reply machine_common_sense/controller.py:313]: Ah, yeah, I changed that when I was debugging. I can change it back, though maybe it's better this way? This condition makes `step()` return `None`, which then might cause the calling script to fail for confusing reasons, which is a reason to keep the change... but I'm not sure.

File: machine_common_sense/controller.py [hunk 2]
```diff
@@ -238,6 +238,18 @@ def start_scene(self, config_data: Union[SceneConfiguration, Dict]) -> \
 
         return output
 
+    def _compare_param_values(self, value_1: Any, value_2: Any) -> bool:
```
Comment (deanwetherby) [inline machine_common_sense/controller.py:241]: Is it possible to move this function to the parameter module so as not to clutter the controller?

File: machine_common_sense/controller.py (hunk 1 above)
Comment (kdrumm) [reply machine_common_sense/controller.py:313]: It seems more correct to be an error log, but I prefer not seeing it in integration tests.  I'm fine with either.

Comment (brianpippin) [reply machine_common_sense/controller.py:313]: I agree with Kyle's last statement.  Seems more correct as an error would be good to not appear in the tests.

File: machine_common_sense/controller.py (hunk 2 above)
Comment (brianpippin) [reply machine_common_sense/controller.py:241]: It would be good to move it to the parameter module now so we don't need to refactor it later imo

Comment (ThomasSchellenbergNextCentury) [reply machine_common_sense/controller.py]: Great idea! Not only did I move the function over, but I also added unit tests (now that it's easier to test), which exposed a bug, that I then fixed.

File: machine_common_sense/controller.py (hunk 1 above)
Comment (ThomasSchellenbergNextCentury) [reply machine_common_sense/controller.py]: I ended up changing these back to warning. I figure most people will run using the default logging (that's set to DEBUG), so they'll see the warning message either way.

**Review by deanwetherby — APPROVED** [review]
Thanks for moving the function to the parameters module and writing unit tests. For the purposes of fixing the integration tests, I'm going to approve this PR. However, there is some fundamental/root issue here as this change seems to go counter to our recent push for raising exceptions. I'm not sure what to do about it quite yet though.

**Pull request merged**